### PR TITLE
Implement "Multiformat" column decorator for table

### PR DESCRIPTION
**Pull request by romaninsh**
Implementation of Multiformat column for the table. It allows to use different formatters depending on cell value. Simple use:

``` php
    $table->addColumn('amount', ['Multiformat', 'callback'=>function($a, $b) {

        if($a['is_invoiced'] > 0) {
            return ['Link', 'invoice', ['invoice_id'=>'id']];
        } elseif (abs($a['is_refunded']) < 50) {
            return ['Template', 'Amount was <b>refunded</b>'];
        }

        return ['Money'];
    }]);
```

Screenshot does not really correspond to example above, but demonstrates 3 different decorators for a same column, including Link+Money, Money and Template.

screen shot 2018-03-21 at 17 00 16

 - Documentation: docs/table.rst
 - Demo: demos/table2.php

### CODE REVIEW

File: demos/table2.php [hunk 1]
```diff
@@ -3,33 +3,70 @@
 date_default_timezone_set('UTC');
 include 'init.php';
 
+$app->add(['Header', 'Table with various headers', 'subHeader'=>'Demonstrates how you can add subheaders, footnotes and other insertions into your data table', 'icon'=>'table']);
```
Comment (DarkSide666) [inline demos/table2.php]: table2 demo page is not included in left side menu.

Comment (DarkSide666) [reply demos/table2.php]: also opening that demo page throws exception
```
Error: Class '\atk4\data\Field\Callback' not found (in C:\wamp64\www\git\atk\ui\vendor\atk4\core\src\FactoryTrait.php:153)
#0 C:\wamp64\www\git\atk\ui\vendor\atk4\data\src\Model.php(373): atk4\data\Model->factory(Array, Array, '\\atk4\\data\\Fiel...')
#1 C:\wamp64\www\git\atk\ui\demos\table2.php(49): atk4\data\Model->addField('amount_copy', Array)
#2 {main}
```
If it doesn't do that on your environment, then try running `composer update`

File: src/TableColumn/Multiformat.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<?php
```
Comment (DarkSide666) [inline src/TableColumn/Multiformat.php]: Please add at least some type-hint comments for properties and methods :)

File: demos/table2.php (hunk 1 above)
Comment (romaninsh) [reply demos/table2.php]: fixed both.

File: src/TableColumn/Multiformat.php (hunk 2 above)
Comment (romaninsh) [reply src/TableColumn/Multiformat.php]: adding..

**Pull request merged**